src/router/router.js

```diff
@@ -107,9 +111,19 @@ export const navigate = async function () {
   if (this.parent[symbols.routes]) {
     const previousRoute = currentRoute
     const hash = getHash()
-    const route = matchHash(hash, this.parent[symbols.routes])
-
+    let route = matchHash(hash, this.parent[symbols.routes])
+    let beforeHookOutput
     if (route) {
+      if (route.hooks) {
+        if (route.hooks.before) {
+          beforeHookOutput = await route.hooks.before(route, previousRoute)
+          if (isString(beforeHookOutput)) {
+            to(beforeHookOutput)
```

Comment:
The redirect works great! One small thing is that we shouldn't store the route we are redirecting from in history, because then the back functionality is broken.